ada_feeding/ada_feeding/trees/acquire_food_tree.py

```diff
@@ -82,9 +86,96 @@ def create_tree(
     ) -> py_trees.trees.BehaviourTree:
         # Docstring copied by @override
 
+        # pylint: disable=line-too-long
+        # This is the entire tree rolled out.
+
         # TODO: remove tree_root_name
         # Sub-trees in general should not need knowledge of their parent.
 
+        ### Blackboard Constants
+        blackboard = py_trees.blackboard.Client(name=name, namespace=name)
+        blackboard.register_key(key="zero_twist", access=py_trees.common.Access.WRITE)
+        blackboard.zero_twist = TwistStamped(
+            header=Header(
+                stamp=self._node.get_clock().now().to_msg(),
+                frame_id="world",
+            ),
+            twist=Twist(),
+        )
+
+        ### Define Tree Leaf Nodes
```

Comment:
What was the motivation for defining these leaf nodes separately, but the other leaf nodes as part of the XML-like definition?

I think having these leaf nodes be separate is a bit confusing to read since it is unclear where variables like `approach_thresh` get written to the blackboard. Whereas if you define these leaf nodes in the tree structure, then readers would already have seen that `Compute...` outputs those blackboard variables.

Reply:
Incorporated into tree.